src/sentry/models/dynamicsampling.py

```diff
@@ -0,0 +1,258 @@
+                        existing_rule.projects.clear()
+                    else:
+                        # add the new projects to the rule, if not already there
+                        for project_id in project_ids:
+                            CustomDynamicSamplingRuleProject.objects.get_or_create(
+                                custom_dynamic_sampling_rule=existing_rule,
+                                project_id=project_id,
+                                organization_id=organization_id,
+                            )
+                # for org rules we don't need to do anything with the projects
+                existing_rule.save()
+                return existing_rule
+            else:
+                # create a new rule
+                condition_hash = get_condition_hash(condition)
+                is_org_level = len(project_ids) == 0
+                condition_str = json.dumps(condition)
+                rule = CustomDynamicSamplingRule.objects.create(
+                    organization_id=organization_id,
+                    condition=condition_str,
+                    sample_rate=sample_rate,
+                    start_date=start,
+                    end_date=end,
+                    num_samples=num_samples,
+                    condition_hash=condition_hash,
+                    is_active=True,
+                    is_org_level=is_org_level,
+                )
+
+                rule.save()
+                # now try to assign a rule id
+                id = rule.assign_rule_id()
+                if id > MAX_CUSTOM_RULES:
+                    # we have too many rules, delete this one
+                    rule.delete()
+                    raise TooManyRules()
+
+                # set the projects if not org level
+                for project_id in project_ids:
+                    CustomDynamicSamplingRuleProject.objects.get_or_create(
+                        custom_dynamic_sampling_rule=rule,
+                        project_id=project_id,
+                        organization_id=organization_id,
+                    )
+
+                return rule
+
+    def assign_rule_id(self) -> int:
+        """
+        Assigns the smallest rule id that is not taken in the
+        current organization.
+        """
+        table_name = self._meta.db_table
+        if self.id is None:
+            raise ValueError("Cannot assign rule id to unsaved object")
+        if self.rule_id != 0:
+            raise ValueError("Cannot assign rule id to object that already has a rule id")
+
+        now = timezone.now()
+
+        raw_sql = (
+            f"UPDATE {table_name} SET rule_id = ( "
+            f"   SELECT COALESCE ((SELECT MIN(rule_id) + 1  FROM {table_name} WHERE rule_id + 1 NOT IN ("
+            f"       SELECT rule_id FROM {table_name} WHERE organization_id = %s AND end_date > %s AND "
+            f"is_active)),1))  "
+            f"WHERE id = %s"
+        )
+        with connections["default"].cursor() as cursor:
```

Comment:
```suggestion
        from django.db import connection
        with connection.cursor() as cursor:
```

Reply:
yes it's the same, but since we do have more databases I like to be explicit that I use the "default" one.